src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -349,4 +375,11 @@ private static void sendResponseMessages(RelayedHttpListenerContext context, boo
 		}
 		return bytes;
 	}
+	
+	/** Validates if the ByteBuffer to be sent was modified after being sent **/
+	private static void checkMsgToSendIsModified(ByteBuffer msgToSend, ByteBuffer afterSend) {
```

Comment:
>checkMsgToSendIsModified [](start = 21, length = 24)

Shouldn't the position move to limit as a result of being sent?

Reply:
It should, but since we are deepcopying the buffer, the original is untouched during sending. Unless we manually change the position/remaining to simulate the effects

---
In reply to: [296937569](https://github.com/Azure/azure-relay-java/pull/48#discussion_r296937569) [](ancestors = 296937569)

Reply:
I realize I have been sending mixed signals here. A general method named deepCopy probably shouldn't consume the original, but a send method which internally does a deep copy to work-around a bad behavior in the WebSocket class probably should mark the ByteBuffer that is passed in as being consumed.

---
In reply to: [296938073](https://github.com/Azure/azure-relay-java/pull/48#discussion_r296938073) [](ancestors = 296938073,296937569)